tests/core/test_save_restore.py

```diff
@@ -549,6 +670,339 @@ def test_mock_model_model_collision(self):
+                parent_path = os.path.join(tmpdir_parent, "parent.nemo")
+                with tempfile.TemporaryDirectory() as tmpdir_child_with_child:
+                    child_with_child_path = os.path.join(tmpdir_child_with_child, 'child_with_child.nemo')
+                    with tempfile.TemporaryDirectory() as tmpdir_child:
+                        # save child
+                        child_path = os.path.join(tmpdir_child, 'child.nemo')
+                        child.save_to(child_path)
+
+                        # create child model with child
+                        cfg_child_with_child = _mock_model_with_children_config(
+                            child1_model_path=None, child2_model_path=child_path
+                        )
+                        cfg_child_with_child.model.temp_file = file_child_with_child.name
+                        child_with_child = MockModelWithChildren(cfg_child_with_child.model)
+                        child_with_child.save_to(child_with_child_path)
+                    # create parent model with child-with-child, leaf checkpoint is not available here
+                    cfg_parent = _mock_model_with_children_config(
+                        child1_model_path=None, child2_model_path=child_with_child_path
+                    )
+                    cfg_parent.model.temp_file = file_parent.name
+                    parent = MockModelWithChildren(cfg_parent.model)
+                    parent.save_to(parent_path)
+
+                # restore, separate children checkpoints are not available here
+                # tmpdir_child, tmpdir_child_with_child are destroyed
+                parent = ModelPT.restore_from(parent_path)
+                # model is transparent, children and model itself can be saved/restored
+                child = self.__test_restore_elsewhere(parent.child2_model.child2_model, map_location='cpu')
+                child_with_child = self.__test_restore_elsewhere(parent.child2_model, map_location='cpu')
+                parent = self.__test_restore_elsewhere(parent, map_location='cpu')
+
+                # test resources for all restored models
+                # leaf model
+                assert child.temp_data == child_data
+                # child with child
+                assert child_with_child.temp_data == child_with_child_data
+                assert child_with_child.child2_model.temp_data == child_data
+                # parent
+                assert parent.temp_data == parent_data
+                assert parent.child2_model.temp_data == child_with_child_data
+                assert parent.child2_model.child2_model.temp_data == child_data
+
+    @pytest.mark.unit
+    @pytest.mark.with_downloads
+    def test_mock_model_nested_child_from_pretrained(self):
+        """
+        Test nested model with child initialized from pretrained model
+        """
+        cfg = _mock_model_with_child_encdecctc_config("QuartzNet15x5Base-En")
```

Comment:
Quartznet doesnt have any artifact, switch to `stt_en_conformer_ctc_small`

Reply:
Done